bot_impl.py

```diff
@@ -5620,20 +5634,26 @@ async def on_message(message):
 
                     vt = int(argument == "yes" or argument == "y")
 
+                    if voudon_in_play and voudon_in_play != person and not person.is_ghost and vt > 0:
+                        await safe_send(message.author, "Voudon is in play. Only the Voudon and dead may vote.")
+                        return
+
                     await vote.vote(vt, operator=message.author)
                     if global_vars.game is not NULL_GAME:
                         backup("current_game.pckl")
                     return
 
-                if (
-                    vote.order[vote.position].user
-                    != (await get_player(message.author)).user
-                ):
+                voting_player = (await get_player(message.author))
+                if vote.order[vote.position].user != voting_player.user:
                     await safe_send(message.author, "It's not your vote right now. Do you mean @presetvote?")
                     return
 
                 vt = int(argument == "yes" or argument == "y")
 
+                if voudon_in_play and voting_player != voudon_in_play and not voting_player.is_ghost and int(argument == "yes" or argument == "y") > 0:
```

Comment:
this as well would be updated to checking

```py
player.character.role_name != "Voudon"
```

Reply:
Also, maybe I'm missing so weird edge case, but why are you casting to an int and comparing to zero rather than just using the booleans directly?

```py
if voudon_in_play and voting_player != voudon_in_play and not voting_player.is_ghost and (argument == "yes" or argument == "y"):
```